lib/elasticsearch/api/http.ex

```diff
@@ -2,91 +2,43 @@ defmodule Elasticsearch.API.HTTP do
   @moduledoc """
   A "real" HTTP implementation of `Elasticsearch.API`.
   """
+  use CarReq
 
   @behaviour Elasticsearch.API
 
   @impl true
   def request(config, method, url, data, opts) do
-    method
-    |> HTTPoison.request(
-      process_url(url, config),
-      process_request_body(data, config),
-      headers(config),
-      opts ++ Map.get(config, :default_options, [])
-    )
-    |> process_response(config)
-  end
-
-  # Respect absolute URLs if passed
-  defp process_url("http" <> _rest = url, _config) do
-    url
-  end
-
-  # On relative urls, prepend the configured base URL
-  defp process_url(url, config) do
-    Path.join(config.url, url)
+    [
+      base_url: Map.get(config, :url),
+      method: method,
+      url: url,
+      headers: Map.get(config, :default_headers, [])
+    ]
+    |> Keyword.merge(process_request_body(data))
```

Comment:
`process_request_body/2` returns `[body:, headers:]`. `Keyword.merge` will overwrite `:headers` set at line 15 with the headers returned from this function.

Sk ce